design/replace-CLI-with-SDK.md

```diff
@@ -0,0 +1,164 @@
+necessary steps to make the Kopia server ready using Kopia SDK and Kopia CLI. 
+
+## High Level Design
+
+### Kopia SDK Wrappers
+
+#### Storage pkg
+
+```go
+
+package storage
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/repo/blob"
+)
+
+type StorageType string
+
+const (
+	TypeS3        StorageType = "S3"
+	TypeAzure     StorageType = "Azure"
+	TypeFileStore StorageType = "FileStore"
+	TypeGCP       StorageType = "GCP"
+)
+
+type Storage interface {
+	Connect() (blob.Storage, error)
+	SetOptions(context.Context, map[string]string)
+	WithCreate(bool)
+}
+
+func New(storageType StorageType) Storage {
+	switch storageType {
+	case TypeS3:
+		return &s3Storage{}
+	case TypeFileStore:
+		return &fileSystem{}
+	case TypeAzure:
+		return &azureStorage{}
+	case TypeGCP:
+		return &gcpStorage{}
+	default:
+		return nil
+	}
+}
+
+```
+
+#### Repository pkg
+
+This pkg would be a wrapper over the `storage pkg` built above and the Kopia repositoy pkg `github.com/kopia/kopia/repo` 
+provided by Kopia SDK
+
+```go
+
+package repository
+
+type Repository struct {
+	st          storage.Storage
+	password    string
+	configFile  string
+	storageType storage.StorageType
+}
+
+// Create repository using Kopia SDK
+func (r *Repository) Create(opts *repo.NewRepositoryOptions) (err error) {
+	storage, err := r.st.Connect()
+	if err != nil {
+		return err
+	}
+	return repo.Initialize(context.Background(), storage, opts, r.password)
+}
+
+// Connect to the repository using Kopia SDK
+func (r *Repository) Connect(opts *repo.ConnectOptions) (err error) {
+	storage, err := r.st.Connect()
+	if err != nil {
+		return err
+	}
+	return repo.Connect(context.Background(), r.configFile, storage, r.password, opts)
+}
+
+// Connect to the repository by providing a config file
+func (r *Repository) ConnectUsingFile() error {
+	repoConfig := repositoryConfigFileName(r.configFile)
+	if _, err := os.Stat(repoConfig); os.IsNotExist(err) {
+		return errors.New("failed find kopia configuration file")
+	}
+
+	_, err := repo.Open(context.Background(), repoConfig, r.password, &repo.Options{})
+	return err
+}
+
+
+func repositoryConfigFileName(configFile string) string {
+	if configFile != "" {
+		return configFile
+	}
+	return filepath.Join(os.Getenv("HOME"), ".config", "kopia", "repository.config")
+}
+
+```
+
+### Repository server controller changes
+
+#### Kopia CLI Approach
+
+![Alt text](images/kopia-CLI-workflow.png?raw=true "Kanister Kopia Integration using Kopia CLI")
+
+Above diagram explains the current workflow repository server controller
+uses to start the Kopia repository server. All the commands are executed from the controller pod inside 
+the respoitory server pod using `kube.exec`. The repository server pod that is created by controller
+uses `kanister-tools` image
+
+
+#### Kopia SDK Approach
+
+![Alt text](images/kopia-SDK-workflow.png?raw=true "Kanister Kopia Integration using Kopia SDK")
```

Comment:
[nit] Maybe we can use different numbering patter (e.g a, b, c, d) for points in "Kopia API server execution" block to avoid numbering sequence with CR workflow